pgjdbc/src/test/java/org/postgresql/test/jdbc2/DatabaseMetaDataTest.java

```diff
@@ -263,7 +263,7 @@ void tables(BinaryMode binaryMode) throws Exception {
 
     rs.close();
 
-    rs = dbmd.getColumns("", "", "meta%", "%");
```

Comment:
Searching with an empty catalog and schema is valid. Please add another test instead of just changing this one

Reply:
@davecramer Per https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getColumns-java.lang.String-java.lang.String-java.lang.String-java.lang.String- : 
> Parameters:
catalog - a catalog name; must match the catalog name as it is stored in the database; "" retrieves those without a catalog; null means that the catalog name should not be used to narrow the search
schemaPattern - a schema name pattern; must match the schema name as it is stored in the database; "" retrieves those without a schema; null means that the schema name should not be used to narrow the search

Searching with an empty catalog and schema should return no results. If you don't want the change in behaviour for how empty strings are handled, I can remove it, because it's not a priority for me/us. However, it technically violates the JDBC spec.

Reply:
I didn't say I wanted to change the behaviour; I said I wanted the test in the same place as the original.

Reply:
@davecramer I separated out the tests for empty vs null.